packages/server/graphql/mutations/createPoll.ts

```diff
@@ -0,0 +1,129 @@
+import {GraphQLNonNull} from 'graphql'
+import {getUserId, isTeamMember} from '../../utils/authorization'
+import publish from '../../utils/publish'
+import CreatePollPayload from '../types/CreatePollPayload'
+import {SubscriptionChannel} from 'parabol-client/types/constEnums'
+
+import {GQLContext} from '../graphql'
+import CreatePollInput from '../types/CreatePollInput'
+import MeetingMemberId from 'parabol-client/shared/gqlIds/MeetingMemberId'
+import insertPoll from '../../postgres/queries/insertPoll'
+import insertPollOptions from '../../postgres/queries/insertPollOptions'
+import segmentIo from '../../utils/segmentIo'
+
+const MAX_POLL_OPTIONS = 4
+const MIN_POLL_OPTIONS = 2
+const MAX_TITLE_LENGTH = 100
+
+type PollOptionsInputVariables = {
+  title: string
+}
+
+type CreatePollInputVariables = {
+  newPoll: {
+    teamId: string
+    discussionId: string
+    title: string
+    threadSortOrder: number
+    options: PollOptionsInputVariables[]
+  }
+}
+
+const createPoll = {
+  type: GraphQLNonNull(CreatePollPayload),
+  args: {
+    newPoll: {
+      type: new GraphQLNonNull(CreatePollInput),
+      description: 'The new poll including title and poll options'
+    }
+  },
+  resolve: async (
+    _source,
+    {newPoll}: CreatePollInputVariables,
+    {authToken, dataLoader, socketId: mutatorId}: GQLContext
+  ) => {
+    const viewerId = getUserId(authToken)
+    const operationId = dataLoader.share()
+    const subOptions = {mutatorId, operationId}
+    const {teamId, discussionId, threadSortOrder, title, options} = newPoll
+
+    //AUTH
+    if (!isTeamMember(authToken, teamId)) {
```

Comment:
can remove, we'll check `Discussion` for this

Reply:
Hmm, not sure I understand. Should we remove this check entirely or read `teamId` from the discussion and then perform the check? If we're going to remove this entirely, how can we ensure that user with a given access token has the permission to add polls to a given discussion? 🤔

Reply:
I guess the answer is reading `teamId` from the discussion, correct me if I'm wrong, please!